### DISCUSSION TITLE
Replace a string with its hash value.

### DISCUSSION THREAD

**Discussion by pollenflugkalender** [BurntSushi/ripgrep#1944] [Q&A] [answered]
Hi guys

I would like to replace sensitive data in logs by replacing the string in question with its hash value.  Example:

_phone number: +4411223344_

should become 

_phone number: 227134d33634fd0c_ 

Is there any way to do this with ripgrep?
And is it maybe also possible to log what was replaced by what.

Kind regards
Polly

**Comment by BurntSushi** [comment, marked as answer]
No, it is not possible and likely never will be. You'll need to use some other tooling to do this. ripgrep is a search tool. It has a `-r/--replace` flag that permits you to do trivial replacements, but does not let you do a replacement using arbitrary code.

**Reply by pollenflugkalender** [reply]
Thank you for the clear answer!
Best regards
Polly
